src/deque.jl

```diff
@@ -74,18 +73,40 @@ type Deque{T}
     Deque() = Deque{T}(DEFAULT_DEQUEUE_BLOCKSIZE)
 end
 
+"""
+    deque(::Type{T})
+
+Create a deque of type `T`
```

Comment:
The signature could be written in a nicer way as `deque(T)` (even if under the hood we need a more complex one).